variadics/src/lib.rs

```diff
@@ -290,17 +313,39 @@ impl RefVariadic for () {
 
 /// A variadic where each item is an exclusive reference `&mut item`.
 ///
+/// This can be created using [`VariadicExt::as_mut_var`]:
+/// ```rust
+/// # use variadics::*;
+/// let as_mut: var_type!(&mut u32, &mut String, &mut bool) =
+///     var_expr!(1_u32, "Hello".to_owned(), false).as_mut_var();
+/// ```
+///
 /// This is a sealed trait.
 #[sealed]
 pub trait MutVariadic: Variadic {
     /// The un-referenced variadic. Each item will have one layer of exclusive references removed.
     ///
     /// The inverse of [`VariadicExt::AsMutVar`].
+    ///
+    /// ```rust
+    /// # use variadics::*;
+    /// let un_mut: <var_type!(&mut u32, &mut String, &mut bool) as MutVariadic>::UnMut =
+    ///     var_expr!(1_u32, "Hello".to_owned(), false);
+    /// ```
     type UnMut: VariadicExt;
 
     /// Type returned by [`Self::downgrade`].
     type Downgrade: RefVariadic<UnRef = Self::UnMut>;
     /// Downgrade this to a variadic of shared reference [`RefVariadic`].
+    ///
+    /// ```rust
+    /// # use variadics::*;
+    /// let mut original = var_expr!(1_u32, "Hello".to_owned(), false);
+    /// let as_mut: var_type!(&mut u32, &mut String, &mut bool) = original.as_mut_var();
+    /// let as_ref_1: var_type!(&u32, &String, &bool) = as_mut.downgrade();
+    /// let as_ref_2: var_type!(&u32, &String, &bool) = as_ref_1; // Can copy the reference version.
```

Comment:
maybe pick a term that references mutability or its opposite? "downgrade" is quite vague. how about `_as_immut_var()`?